app/components/post_body/system_message_helpers.js

```diff
@@ -0,0 +1,180 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {Posts} from 'mattermost-redux/constants';
+import Markdown from 'app/components/markdown';
+
+const renderUsername = (value) => {
+    return (value[0] === '@') ? value : `@${value}`;
+};
+
+const formattedMessage = (localeHolder, elements, intl) => {
+    const {formatMessage} = intl;
+
+    return formatMessage(localeHolder, elements);
+};
+
+const renderHeaderChangeMessage = (postBodyProps, styles, intl) => {
+    const {onPress, postProps} = postBodyProps;
+    const {messageStyle, textStyles} = styles;
+
+    if (!postProps.username) {
+        return null;
+    }
+
+    const username = renderUsername(postProps.username);
+    const oldHeader = postProps.old_header;
+    const newHeader = postProps.new_header;
+    let localeHolder;
+
+    if (postProps.new_header) {
+        if (postProps.old_header) {
+            localeHolder = {
+                id: 'api.channel.post_update_channel_header_message_and_forget.updated_from',
```

Comment:
you need to import `import {t} from 'app/utils/i18n` and then do 
```suggestion
                id: t('api.channel.post_update_channel_header_message_and_forget.updated_from'),
```

this is the reason why it is being removed from the en.json file.

Also the ids are kind of what the server uses, I would rather have it as `mobile.system_message.update_channel_header` or something like that. The same applies to all others.

Reply:
These same strings are used by the web app as well, so I don't think we need to change the IDs right now

Reply:
I'm more in favour of prefixing these with `mobile` to follow the majority pattern for i18n string keys in mobile. There is only one existing i18n string key that starts with `api` and I don't think I want to add any more pollution. Will change the key suffix as per suggestion.

Reply:
Changes and renaming done as per suggestion.